internal/gatewayapi/helpers_v1alpha2.go

```diff
@@ -44,43 +44,6 @@ func PortNumPtrV1Alpha2(port int) *gwapiv1a2.PortNumber {
 	return &pn
 }
 
-func UpgradeParentReferences(old []gwapiv1a2.ParentReference) []gwapiv1.ParentReference {
```

Comment:
hey is this refactor related to this GH issue, if not, can we seperate it from this PR ?

Reply:
hey still seeing these refactors, so unresolving this one

Reply:
The updates in `GetRouteParentContext` and `GetParentReferences` is necessary. `UpgradeParentReferences` will do a deepcopy for the pointer fields and makes `ParentReference`s unable to used as `map` keys.

Reply:
cc @shawnh2 who had last updated this lib

Reply:
Sorry for the delay. The updates in `GetRouteParentContext` and `GetParentReferences` is indeed necessary, since  the `ParentRef` for xRoute got all upgraded to `v1`.

So the codes here do need a clean-up, but we can track this in a separate issue.